llarp/aligned.hpp

```diff
@@ -57,75 +73,72 @@ namespace llarp
     operator~() const
     {
       AlignedBuffer< sz > ret;
-      size_t idx = 0;
-      while(idx < sz / sizeof(Long_t))
-      {
-        ret.data_l()[idx] = ~data_l()[idx];
-        ++idx;
-      }
+      std::transform(as_array().begin(), as_array().end(),
+                     ret.as_array().begin(), [](byte_t a) { return ~a; });
+
       return ret;
     }
 
     bool
     operator==(const AlignedBuffer& other) const
     {
-      return memcmp(b, other.b, sz) == 0;
+      return as_array() == other.as_array();
     }
 
     bool
     operator!=(const AlignedBuffer& other) const
     {
-      return !(*this == other);
+      return as_array() != other.as_array();
     }
 
     bool
     operator<(const AlignedBuffer& other) const
     {
-      return memcmp(l, other.l, sz) < 0;
+      return as_array() < other.as_array();
     }
 
     bool
     operator>(const AlignedBuffer& other) const
     {
-      return memcmp(l, other.l, sz) > 0;
+      return as_array() > other.as_array();
     }
 
     bool
     operator<=(const AlignedBuffer& other) const
     {
-      return memcmp(l, other.l, sz) <= 0;
+      return as_array() <= other.as_array();
     }
 
     bool
     operator>=(const AlignedBuffer& other) const
     {
-      return memcmp(l, other.l, sz) >= 0;
+      return as_array() >= other.as_array();
     }
 
     AlignedBuffer
     operator^(const AlignedBuffer& other) const
     {
       AlignedBuffer< sz > ret;
-      for(size_t idx = 0; idx < sz / sizeof(Long_t); ++idx)
-        ret.l[idx] = l[idx] ^ other.l[idx];
+      std::transform(as_array().begin(), as_array().end(),
```

Comment:
the point of doing it with longs is so that it's optimized by the compiler better

Reply:
https://godbolt.org/z/FNhPK7 <- wildly easy to optimise.

Reply:
doing it with `-mach=native` just uses AVX instructions - https://godbolt.org/z/Uei2HP